.github/workflows/main.yml

```diff
@@ -0,0 +1,38 @@
+on: [push, pull_request]
+
+name: Build and Deploy
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Install requirements
+      run: |
+        pip install -r beginners/requirements.txt
+
+    - name: Build GitHub pages
+      run: |
+        cd beginners
+        make github
+
+    - name: Deploy tutorial on the cvc5 website
+      if: (github.repository == 'cvc5/tutorials') && (github.ref == 'refs/heads/main')
+      env:
+        SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+      run: |
+        ssh-agent -a $SSH_AUTH_SOCK > /dev/null
+        ssh-add - <<< "${{ secrets.CVC5_WEBSITE_TOKEN }}"
+
+        git config --global user.email "webbot@cvc5"
+        git config --global user.name "WebBot"
```

Comment:
Maybe change this to cvc5-bot since we use the cvc5 bot account for pushing changes.

Reply:
Done